### ISSUE TITLE
Possible False Positive Virus Detection in Patchwork-3.10.1-windows.exe

### ISSUE THREAD

**Issue by rolandog** [ssbc/patchwork#855]
Hi. I'm new to the SSB ecosystem.

I had been hesitant to download the executable as I couldn't (easily) find a cryptographic hash (like SHA256) or signature file and public key to validate that the file I received corresponds to the one published by the developers.

I just downloaded `Patchwork-3.10.1-windows.exe` (step 1 of the 'How to get started' guide is to install Patchwork), and -- out of precaution, as I couldn't find the aforementioned SHA256 hash and .sig files -- I decided to lookup the downloaded file's hash in VirusTotal.

The hash I got from downloading the windows installer (both from GitHub and from the link in the guide) is: `92603a3e099d9517e1db17f8dce25ebb56d26e7a2c5581ed95bff0e2a8025409`

I was surprised that VirusTotal [reported two different hits](https://www.virustotal.com/en/file/92603a3e099d9517e1db17f8dce25ebb56d26e7a2c5581ed95bff0e2a8025409/analysis/):

* Bkav reported `HW32.Packed.7D8F`
* Comodo reported `TrojWare.JS.Agent.ANI`

I scanned `Patchwork-3.10.0-windows.exe`, with SHA256 `8dd8707be9f5d8a23e237a3a1f67c747076962ed5d47c493bc8860d42429cdd3`, just in case, and it [only received one positive result](https://www.virustotal.com/en/file/8dd8707be9f5d8a23e237a3a1f67c747076962ed5d47c493bc8860d42429cdd3/analysis/), this time only by Bkav, listing it as `HW32.Packed. `.

I'm inclined to say that maybe Bkav is showing a False Positive, but I don't know about Comodo.

I checked the Mac and Linux versions in VirusTotal and both were reported to be clean.

Also, I found a similar report on a [different project](https://github.com/beakerbrowser/beaker/issues/898); it seemed like one of the cases was a false positive due to the installer not being signed.

**TL;DR:**

1. Can the developers validate if the file is indeed virus-free?
2. Also, can anyone point me in the right direction as to where I may find SSBC's public keys and / or cryptographic hashes to validate proper file integrity before installing?

**Comment by rolandog** [comment]
I found further information pointing out that, [Bkav's result is probably a false positive](https://old.reddit.com/r/qBittorrent/comments/89n5bx/virus_total_detected_this_engine/).

Does anyone know if `TrojWare.JS.Agent.ANI` detections by Comodo are false positives?

**Comment by rolandog** [comment]
Thanks for your prompt response @christianbundy !

The only pending issue would be to confirm if the difference in the antivirus results for the Windows versions (from 3.10.0 to 3.10.1, where the latter is reported to have `TrojWare.JS.Agent.ANI`) are real or if they're false positives as well.

From one of the external issues I linked, a commenter made reference to F-Secure's details on [Trojan-Downloader:JS/Agent](https://www.f-secure.com/v-descs/trojan-downloader_js_agent.shtml) and, particularly, on [Trojan-Downloader:JS/Agent.ANI](https://www.f-secure.com/v-descs/trojan-downloader_js_agent_ani.shtml); those infections seem to be about something completely different (buffer overflow exploits of an ActiveX control). It would seem to me to be a false positive. On the other hand, the commenter also linked to [this McAfee](https://securingtomorrow.mcafee.com/mcafee-labs/malware-hides-in-installer-to-avoid-detection/) post on malware hiding in NSIS installers.

This last point is the one that would be interesting to address, as the previous Windows installer of Patchwork didn't raise any `TrojWare.JS.Agent.ANI` flags when scanned in VirusTotal.

**Comment by mmckegg** [comment]
We're now up to version 3.11.4, so I'm going to close this. Feel free to reopen if the false positives reappear!
